OpenRA.Mods.D2k/Activities/SwallowActor.cs

```diff
@@ -107,10 +107,14 @@ void PlayAttack(Actor self, WPos attackPosition, List<Player> affectedPlayers)
 			withSpriteBody.PlayCustomAnimation(self, sandworm.Info.MouthSequence);
 			Game.Sound.Play(swallow.Info.WormAttackSound, self.CenterPosition);
 
+			if (affectedPlayers.Count == 0)
+				return;
+
 			Game.RunAfterDelay(1000, () =>
 			{
-				foreach (var affectedPlayer in affectedPlayers)
-					NotifyPlayer(affectedPlayer, attackPosition);
+				if (Game.IsCurrentWorld(affectedPlayers[0].World))
```

Comment:
The only reason `affectedPlayers[0].World` is used is to get the `World` containing it which is the same for all in `affectedPlayers`, making the result the same for any in `affectedPlayers`.

This code should be functionally equivalent:
```c#
 			var world = Game.OrderManager.World;
 			Game.RunAfterDelay(1000, () =>
 			{
				if (Game.IsCurrentWorld(world))
					foreach (var affectedPlayer in affectedPlayers)
						NotifyPlayer(affectedPlayer, attackPosition);
 			});
```